tests/unit/BackendClient-spec.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * Created by Ron.Yadgar on 11/3/2015.
+ */
+var proxyquire = require('proxyquire');
+var chai = require('chai');
+var expect = chai.expect;
+var sinon = require('sinon');
+var config = require('../../common/Configuration');
+var assert=require('assert')
+describe('Backend client facdefgstory spec', function() {
+    var loggerMock = {
+        info: sinon.stub(),
+        error: sinon.stub(),
+        debug: sinon.stub(),
+        warn: sinon.stub()
+    };
+    var createBackendClient = function (mocks) {
+        var BackendClientCtor = proxyquire('../../lib/BackendClient', mocks);
+        return BackendClientCtor;
+    };
+
+    it("Session expired, check for creating new session", function (done) {
+        KalturaClientMock = {
+            KalturaClient: sinon.stub().returns({
+                session: {
+                    start: sinon.stub().callsArgWith(0, '1')
+                },
+                setSessionId: sinon.stub(),
+                liveStream: {
+                    listAction: sinon.stub().callsArgWith(0,'1')
+                }
+            })
+        };
+        var mocks = {
+            './logger/logger': sinon.stub().returns(loggerMock),
+            './kaltura-client-lib/KalturaClient': KalturaClientMock
+        };
+        var sessionDurationTest = config.get('backendClient').ksSessionRefreshIntervalMinutes * 60 * 1000;
+        var fakeTime = sessionDurationTest + 1;
+        clock = sinon.useFakeTimers(fakeTime);
+        var BackendClient = createBackendClient(mocks);
+        BackendClient.SetLastSessionTime(0);
+        BackendClient.getLiveEntriesForMediaServer().then(
+            function () {
+                console.log(BackendClient.GetlastSessionTime());
+                console.log('Promise Succsed');
+                expect(BackendClient.GetlastSessionTime()).to.eql(fakeTime);
+                done();
+            },
+            function(err)
+            {
+                console.log('Promise failed');
+                done(err);
+            });
+    });
+    it ("Promise should be failed, after error occurred, in the create new session",function(done){
+        KalturaClientMock = {
```

Comment:
Move this object along with the mocks object to the createBackendClient function since they all pretty much use the same code. In your tests where you wish to override something, do so via a custom function - look at createManifestGenerator(customizeMocks, ... ) in ChunklistManifestGenerator-spec.js as an example.
Then, you can just override the relevant method (e.g. mocks['./someModule'].someStub.returns(newValue);